### PR TITLE
Fixed autotools

### PR DESCRIPTION
**Pull request by brtnfld**
Checked autotool tests with 1.8.22 (with and without F2003), 1.10.8, 1.12.1, develop

@byrnHDF I needed to refine the hdf5 version check, so I added a similar H5_VERSION check as in C.
The new include file for Fortran H5D/h5_version.h.in set the major,minor,release of the HDF5 version. I'm not sure how to do the same with CMake.

Also, I don't think the CMake version will work with 1.8 with F2003 disabled. At least I did not see an F2003 conditonal.

### CODE REVIEW

**Review by lrknox — APPROVED** [review]

File: C/H5D/testfiles/18/h5ex_d_nbit.ddl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 HDF5 "h5ex_d_nbit.h5" {
 GROUP "/" {
    DATASET "DS1" {
-      DATATYPE  32-bit little-endian integer 16-bit precision
+      DATATYPE  32-bit little-endian integer
```
Comment (byrnHDF) [inline C/H5D/testfiles/18/h5ex_d_nbit.ddl:4]: Did 1.8 not get the changes to h5dump that was in the other branches? Do we want to make that change?

File: FORTRAN/H5D/h5_version.h.in [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+!
+#define H5_VERS_MINOR @H5_VERS_MINOR@
+!
+! For tweaks, bug-fixes, or development
+!
+#define H5_VERS_RELEASE @H5_VERS_RELEASE@
+
+! macros for comparing versions
+
+#define H5_VERSION_GE(Maj, Min, Rel)                                                                         \
```
Comment (byrnHDF) [inline FORTRAN/H5D/h5_version.h.in:17]: so these are the fortran versions of what is in the library?

File: FORTRAN/H5D/h5_version.h.in [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+! For minor interface/format changes
+!
+#define H5_VERS_MINOR @H5_VERS_MINOR@
+!
+! For tweaks, bug-fixes, or development
+!
+#define H5_VERS_RELEASE @H5_VERS_RELEASE@
+
+! macros for comparing versions
+
```
Comment (byrnHDF) [inline FORTRAN/H5D/h5_version.h.in:16]: Also, how to handle the !defined(H5_USE_XX_API) usage in fortran?

Comment (brtnfld) [reply FORTRAN/H5D/h5_version.h.in:16]: Fortran does not have any versioned APIs, so from an application's point of view it should not be needed.

File: FORTRAN/H5D/h5_version.h.in (hunk 2 above)
Comment (brtnfld) [reply FORTRAN/H5D/h5_version.h.in:17]: yes, I copied the source from H5public.h

File: C/H5D/testfiles/18/h5ex_d_nbit.ddl (hunk 1 above)
Comment (brtnfld) [reply C/H5D/testfiles/18/h5ex_d_nbit.ddl:4]: I guess not, at least for 1.8.22.
Did you mean to make the change to h5dump in 1.8.23?

File: configure.ac [hunk 4]
```diff
@@ -84,16 +86,25 @@ AC_MSG_RESULT([$H5_APIVER])
...
+H5_RELEASEVER=$(echo $H5_LIBVER | cut -f3 -d'.'  | sed -r 's/\./_/g')
+
+#used in fortran version checks
+H5_VERS_MAJOR=$H5_MAJORVER
+H5_VERS_MINOR=$H5_MINORVER
+H5_VERS_RELEASE=$H5_RELEASEVER
+
+AC_SUBST([H5_VERS_MAJOR])
+AC_SUBST([H5_VERS_MINOR])
+AC_SUBST([H5_VERS_RELEASE])
```
Comment (byrnHDF) [inline configure.ac:101]: CMake from the hdf5 library CMakeLists.txt file:

```
----------------------------------------------------------------------------
 parse the full version number from H5public.h and include in H5_VERS_INFO
-----------------------------------------------------------------------------
file (READ ${HDF5_SRC_DIR}/H5public.h _h5public_h_contents)
string (REGEX REPLACE ".*#define[ \t]+H5_VERS_MAJOR[ \t]+([0-9]*).*$"
    "\\1" H5_VERS_MAJOR ${_h5public_h_contents})
string (REGEX REPLACE ".*#define[ \t]+H5_VERS_MINOR[ \t]+([0-9]*).*$"
    "\\1" H5_VERS_MINOR ${_h5public_h_contents})
string (REGEX REPLACE ".*#define[ \t]+H5_VERS_RELEASE[ \t]+([0-9]*).*$"
    "\\1" H5_VERS_RELEASE ${_h5public_h_contents})

```

File: FORTRAN/H5D/h5_version.h.in (hunk 3 above)
Comment (brtnfld) [reply FORTRAN/H5D/h5_version.h.in:16]: I've not checked the examples with an HDF5 version other than the default API mappings.

File: configure.ac (hunk 4 above)
Comment (brtnfld) [reply configure.ac:101]: I updated to using versioning in Fortran, thanks.

**Comment by brtnfld** [comment]
@byrnHDF, I can't seem to get the canonical hdf5-examples  to work with Fortran:

-- HDF5 C libs:1 static: and shared:1
-- HDF5 Fortran libs: static: and shared:1
-- HDF5 Java libs: 
-- HDF5 Fortran libs not found - disable build of Fortran examples
-- HDF5-1.13.2 found: INC=/home/brtnfld/packages/hdf5/build.cmake/hdf5/include; TOOLS=/home/brtnfld/packages/hdf5/build.cmake/hdf5/bin

**Comment by brtnfld** [comment]
Should I just merge this and address the general Fortran issues in another PR?

**Comment by lrknox** [comment]
Seems good to me.

________________________________

**Comment by byrnHDF** [comment]
> Should I just merge this and address the general Fortran issues in another PR?

Yeah, I can take a look after the merge.

File: C/H5D/testfiles/18/h5ex_d_nbit.ddl (hunk 1 above)
Comment (byrnHDF) [reply C/H5D/testfiles/18/h5ex_d_nbit.ddl:4]: Maybe it was the next 1.8 release. I probably was using current development - so may need to add version checks for the tests?

File: FORTRAN/H5D/h5_version.h.in (hunk 3 above)
Comment (byrnHDF) [reply FORTRAN/H5D/h5_version.h.in:16]: I'll do that afterwards.

File: FORTRAN/H5D/testfiles/110/h5ex_d_transform.ddl [hunk 5]
```diff
@@ -1,17 +1,6 @@
...
-      (1,0): 0, 1, 2, 3,
-      (2,0): -1, 1, 3, 5,
-      (3,0): -2, 1, 4, 7,
-      (4,0): -3, 1, 5, 9,
-      (5,0): -4, 1, 6, 11,
-      (6,0): -5, 1, 7, 13
-      }
-   }
-}
+FILE_CONTENTS {
```
Comment (byrnHDF) [inline FORTRAN/H5D/testfiles/110/h5ex_d_transform.ddl:2]: curious why the difference.

**Review by byrnHDF — APPROVED** [review]
Get this merged then work on any issues.

File: FORTRAN/H5D/testfiles/110/h5ex_d_transform.ddl (hunk 5 above)
Comment (brtnfld) [reply FORTRAN/H5D/testfiles/110/h5ex_d_transform.ddl:2]: In the past, we did not check the numerical difference. I would have to check why we ended up doing that.

**Comment by brtnfld** [comment]
Also, if this requires a cmake-built HDF5 (and it looks like it does), we should check for this with an error message for Autotools HDF5.

**Comment by byrnHDF** [comment]
> Also, if this requires a cmake-built HDF5 (and it looks like it does), we should check for this with an error message for Autotools HDF5.

Should work for autotools built hdf5.

**Comment by brtnfld** [comment]
Where are variables such as HDF5_BUILD_FORTRAN getting set for an Autotools HDF5?

**Comment by byrnHDF** [comment]
Not sure about that, because it is dependent on the FIND_HDF5.cmake file supplied by CMake. It uses the h5cc script for the C settings.

________________________________________

**Pull request merged**